torch/csrc/jit/codegen/cuda/fusion.cpp

```diff
@@ -8,12 +8,30 @@ namespace fuser {
 
 static thread_local Fusion* ACTIVE_FUSION = nullptr;
 
-FusionGuard::FusionGuard(Fusion* fusion) {
+static thread_local bool AUTO_PRINT = false;
+
+FusionGuard::FusionGuard(Fusion* fusion, const char* context_name) : 
```

Comment:
Would context_name make sense as a member of fusion?

Reply:
It seems we may use the same fusion objects in multiple contexts. If that's the case, something like context_name can't be tied to the fusion itself, right?